src/hotspot/share/gc/parallel/psScavenge.cpp

```diff
@@ -703,9 +703,6 @@ void PSScavenge::clean_up_failed_promotion() {
   ParallelScavengeHeap* heap = ParallelScavengeHeap::heap();
   PSYoungGen* young_gen = heap->young_gen();
 
-  RemoveForwardedPointerClosure remove_fwd_ptr_closure;
-  young_gen->object_iterate(&remove_fwd_ptr_closure);
-
```

Comment:
Since these are removed, there is no longer any need to get `heap` and `young_gen` above either. I think the cleanest would be do directly use `ParallelScavengeHeap::heap()` in: 
```
NOT_PRODUCT(heap->reset_promotion_should_fail();)
```